lib/wlib/strings/DynamicString.h

```diff
@@ -0,0 +1,476 @@
+		return m_buffer[0];
+	}
+
+	/**
+	 * Provides access to the last character in the string
+	 *
+	 * @return the last character
+	 */
+	char &end(){
+		if (empty()) return m_buffer[0];
+
+		return m_buffer[length() - 1];
+	}
+
+	/**
+	 * Provides access to the last character in the string. Character is constant
+	 *
+	 * @return the last character
+	 */
+	const char &end() const{
+		if (empty()) return m_buffer[0];
+
+		return m_buffer[length() - 1];
+	}
+
+	/**
+	 * Modifier operator adds character to the current string.
+	 *
+	 * @param c character to add
+	 * @return the current string
+	 */
+	DynamicString &operator+=(char c){
+		char array[1] = {c};
+		return append(array);
+	}
+
+	/**
+	 * Modifier operator adds char string to the current string.
+	 *
+	 * @param val char string to add
+	 * @return the current string
+	 */
+	DynamicString &operator+= (const char* val){
+		return append(val);
+	}
+
+	/**
+	 * Modifier operator adds @code DynamicString object to the current string.
+	 *
+	 * @param other @code DynamicString string to add
+	 * @return the current string
+	 */
+	DynamicString &operator+= (DynamicString &other){
+		return append(other.c_str());
+	}
+
+	/**
+	 * Appends a character to the current string.
+	 *
+	 * @param c character to add
+	 * @return the current string
+	 */
+	DynamicString &append(const char c){
+		char array[2] = {c, '\0'};
+		return append(array);
+	}
+
+	/**
+	 * Appends a character string to the current string.
+	 *
+	 * @param str character string to add
+	 * @return the current string
+	 */
+	DynamicString &append(const char *str){
+		uint16_t bufferLength = this->length();
+		uint16_t newLength = (uint16_t)(bufferLength + strlen(str));
+		m_buffer = (char *)memory_realloc(m_buffer, (newLength + 1) * sizeof(char));
+		
+		for (int i = bufferLength; i < newLength; i++) {
+			m_buffer[i] = str[i-bufferLength];
+		}
+		
+		m_buffer[newLength] = '\0';
+
+		return *this;
+	}
+
+	/**
+	 * Appends a DynamicString string to the current string.
+	 *
+	 * @param str DynamicString string to add
+	 * @return the current string
+	 */
+	DynamicString &append(const DynamicString &str){
+		return append(str.c_str());
+	}
+
+	/**
+	 * Appends a character to the current string.
+	 *
+	 * @param c character to add
+	 * @return the current string
+	 */
+	DynamicString &push_back(const char c){
+		char array[2] = {c, '\0'};
+		return append(array);
+	}
+
+	/**
+     * Deletes the element @p pos from the String
+     *
+     * @param pos position of the element to be deleted
+     * @return the modified String
+     */
+    DynamicString &erase(uint16_t pos = 0){
+		uint16_t len = this->length();
+        if (len == 0 || pos >= len) return *this;
+
+		len--;
+
+        for (uint16_t i = pos; i < len; ++i) {
+            m_buffer[i] = m_buffer[i+1];
+        }
+
+        m_buffer[len] = '\0';
+        return *this;
+    }
+
+    /**
+     * Deletes the last character in the String
+     */
+    void pop_back(){

```

Comment:
Looks like there are some formatting issues here. Do a replace of all tabs in the code with 4 spaces, and run an auto-format if you've got one in your IDE, otherwise, I can check out the code and do it before the merge.